Source/src/components/ComponentTransform.cpp

```diff
@@ -129,22 +135,22 @@ void Hachiko::ComponentTransform::SetLocalPosition(const float3& new_position_lo
 {
     position_local = new_position_local;
 
-    UpdateTransformOfHierarchy(MatrixCalculationMode::GLOBAL_FROM_LOCAL);
+    dirty = true;
```

Comment:
After the comment from baran I've noticed that shis should set dirty to this trnsform and the children. Something like
```
void ComponentTransform::Invalidate() {
	dirty = true;
	for (child : gameobget-> children)
             child->transform->invalidate();
}
```

Reply:
An example of why it needs to be like that:
With current way:
1.  You update parent and only set him to dirty
2. You call child transform from some reason before parent, child does not know that parent has changed and gives u an outdated transform with the old parent values

So the children needs to be dirty if its parent changes so it is updated even if the parent is called afterwards